src/uikit/Input.js

```diff
@@ -1,4 +1,47 @@
+import React from 'react';
 import styled from 'react-emotion';
+import { withTheme } from 'emotion-theming';
+import Row from 'uikit/Row';
+import SearchIcon from 'react-icons/lib/fa/search';
+import FaTimesCircleO from 'react-icons/lib/fa/times-circle';
+
+// this this component should implement the default <input> type from react-dom with some additional
+// props for styling
+export const FilterInput = withTheme(
+  ({
+    theme,
+    LeftIcon = SearchIcon,
+    RightIcon = FaTimesCircleO,
+    className,
+    value,
+    disabled,
+    ...props
+  }) => {
+    const Wrapper = styled(Row)`
```

Comment:
should be declared outside this component so it's not re-instantiated every render